src/components/SearchAutocomplete/index.js

```diff
@@ -225,3 +200,36 @@ export default class SearchAutocomplete extends Component {
     );
   }
 }
+
+
+function mapStateToProps(state) {
+  const surahs = state.surahs.entities;
+  const surahId = state.surahs.current;
+  let lang = 'en';
+
+  if (state.ayahs && state.ayahs.entities && state.ayahs.entities[surahId]) {
+    const ayahs = state.ayahs.entities[surahId];
+    const ayahKey = Object.keys(ayahs)[0];
+
+    if (ayahKey) {
+      const ayah = ayahs[ayahKey];
+
+      if (ayah.content && ayah.content[0] && ayah.content[0].lang) {
+        lang = ayah.content[0].lang;
+      }
+    }
+  }
+
+  return {
+    surahs,
+    lang
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    push: bindActionCreators(push, dispatch)
```

Comment:
From react-redux docs:

```
[mapDispatchToProps(dispatch, [ownProps]): dispatchProps] (Object or Function): If an object is passed, each function inside it will be assumed to be a Redux action creator. An object with the same function names, but with every action creator wrapped into a dispatch call so they may be invoked directly, will be merged into the component’s props. If a function is passed, it will be given dispatch. It’s up to you to return an object that somehow uses dispatch to bind action creators in your own way. (Tip: you may use the bindActionCreators() helper from Redux.) If you omit it, the default implementation just injects dispatch into your component’s props. If ownProps is specified as a second argument, its value will be the props passed to your component, and mapDispatchToProps will be re-invoked whenever the component receives new props.

```